src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java

```diff
@@ -54,7 +63,29 @@ public void inferenceSentence(
         @NonNull final String inputText,
         @NonNull final ActionListener<List<Float>> listener
     ) {
-        inferenceSentences(TARGET_RESPONSE_FILTERS, modelId, List.of(inputText), ActionListener.wrap(response -> {
+        inferenceSentence(modelId, inputText, null, listener);
+    }
+
+    /**
+     * Wrapper around {@link #inferenceSentences} that expected a single input text and produces a single floating
+     * point vector as a response. Supports passing {@link MLAlgoParams} to the inference. If the model is
+     * asymmetric, passing a
+     * {@link org.opensearch.ml.common.input.parameter.textembedding.AsymmetricTextEmbeddingParameters} is
+     * mandatory. This method will check whether the model being used is asymmetric and correctly handle the
+     * parameter, so it's okay to always pass the parameter (even if the model is symmetric).
+     *
+     * @param modelId {@link String}
+     * @param inputText {@link List} of {@link String} on which inference needs to happen
+     * @param mlAlgoParams {@link MLAlgoParams} which will be used to run the inference
+     * @param listener {@link ActionListener} which will be called when prediction is completed or errored out
+     */
+    public void inferenceSentence(
```

Comment:
We need to refactor this method to accept single POJO object that can be built with builder pattern, similar to ml-commons class with algorithm params. Reason - we cannot add new method every time we need to add new parameter to inference ml client.
I'm fine if we do it in a separate PR, please create GH issue and post link here.

Reply:
Okay. Cf. https://github.com/opensearch-project/neural-search/issues/790